app/src/main/java/ch/blinkenlights/android/vanilla/CoverCache.java

```diff
@@ -463,6 +468,19 @@ public Bitmap createBitmap(Song song, long maxPxCount) {
 					}
 				}
 
+				if (inputStream == null && (CoverCache.mCoverLoadMode & CoverCache.COVER_MODE_INLINE) != 0)
+				{
+					android.media.MediaMetadataRetriever mmr = new MediaMetadataRetriever();
+					mmr.setDataSource(song.path);
+
+					byte[] data = mmr.getEmbeddedPicture();
+					if (data != null) {
+						Bitmap bitmap = BitmapFactory.decodeByteArray(data, 0, data.length);
+						return bitmap;
```

Comment:
Returning here completely skips all the resizing: the code would return a wrongly sized picture.

You should construct an inputStream instead from the byte array.

Reply:
changed